### PR TITLE
Call function to change GATT params

### PR DESCRIPTION
**Pull request by niksiboxi**

### CODE REVIEW

File: src/nrf5_sdk15_platform/communication/ruuvi_nrf5_sdk15_communication_ble_gatt.c [hunk 1]
```diff
@@ -853,4 +859,43 @@ rd_status_t ri_gatt_dis_init (const ri_comm_dis_init_t * const p_dis)
...
+ * @retval RD_ERROR_INVALID_PARAM params was not one of supported defaults
+ * @retval RD_ERROR_INVALID_STATE if there is no ongoing GATT connection
+ * @retval Error code from BLE Stack if applicable
+ */
+rd_status_t ri_gatt_params_request (const ri_gatt_params_t params)
+{
+    ret_code_t err_code = NRF_SUCCESS;
+    ble_gap_conn_params_t gap_conn_params;
+    memset (&gap_conn_params, 0, sizeof (gap_conn_params));
+    gap_conn_params.slave_latency = SLAVE_LATENCY;
```
Comment (ojousima) [inline src/nrf5_sdk15_platform/communication/ruuvi_nrf5_sdk15_communication_ble_gatt.c:876]: Slave latency should be selected to match scenario. 0 on TURBO, "something reasonable" on STANDARD, maximum allowed by Apple on LOW_POWER. 

Determining the proper constants takes some research, they should be #define'd in header and reasoning for the value explained in comments

File: src/nrf5_sdk15_platform/communication/ruuvi_nrf5_sdk15_communication_ble_gatt.c [hunk 2]
```diff
@@ -853,4 +859,43 @@ rd_status_t ri_gatt_dis_init (const ri_comm_dis_init_t * const p_dis)
...
+            gap_conn_params.max_conn_interval = 0x0006;
+            break;
+
+        case RI_GATT_STANDARD:
+            gap_conn_params.min_conn_interval = MIN_CONN_INTERVAL;
+            gap_conn_params.max_conn_interval = MAX_CONN_INTERVAL;
+            break;
+
+        case RI_GATT_LOW_POWER:
+            gap_conn_params.min_conn_interval = 0x0C80;
```
Comment (ojousima) [inline src/nrf5_sdk15_platform/communication/ruuvi_nrf5_sdk15_communication_ble_gatt.c:892]: Reasoning for these values?

File: src/nrf5_sdk15_platform/communication/ruuvi_nrf5_sdk15_communication_ble_gatt.c [hunk 3]
```diff
@@ -318,6 +321,9 @@ static void ble_evt_handler (ble_evt_t const * p_ble_evt, void * p_context)
             evt.type = BLE_NUS_EVT_COMM_STOPPED;
             nus_data_handler (&evt);
             m_conn_handle = BLE_CONN_HANDLE_INVALID;
+            err_code = ri_gatt_params_request (RI_GATT_STANDARD);
```
Comment (ojousima) [inline src/nrf5_sdk15_platform/communication/ruuvi_nrf5_sdk15_communication_ble_gatt.c:324]: This is requesting parameter change for invalid change and will error, delete this line

**Review by DG12 — COMMENTED** [review]
Why did you specify intervals in hex ? if  you meant 3200 please use 3200 (not xC80). 
How did you decide on 6 and 3200 ?

File: src/nrf5_sdk15_platform/communication/ruuvi_nrf5_sdk15_communication_ble_gatt.c [hunk 4]
```diff
@@ -97,7 +97,7 @@
 #define SEC_PARAM_MAX_KEY_SIZE           16                                         /**< Maximum encryption key size. */
 
 #ifndef RUUVI_NRF5_SDK15_COMMUNICATION_BLE4_GATT_LOG_LEVEL
-#define RUUVI_NRF5_SDK15_COMMUNICATION_BLE4_GATT_LOG_LEVEL RI_LOG_LEVEL_DEBUG
+#define RUUVI_NRF5_SDK15_COMMUNICATION_BLE4_GATT_LOG_LEVEL RI_LOG_LEVEL_INFO
```
Comment (ojousima) [inline src/nrf5_sdk15_platform/communication/ruuvi_nrf5_sdk15_communication_ble_gatt.c:100]: Restore to DEBUG before merging

File: src/interfaces/communication/ruuvi_interface_communication_ble_gatt.h [hunk 5]
```diff
@@ -71,4 +71,15 @@ rd_status_t ri_gatt_dfu_init (void);
...
+ *
+ * @param[in] params One of preset defaults: RI_GATT_TURBO, RI_GATT_STANDARD, RI_GATT_LOW_POWER.
+ * @retval RD_SUCCESS Parameter update was requested
+ * @retval RD_ERROR_INVALID_PARAM params was not one of supported defaults
+ * @retval RD_ERROR_INVALID_STATE if there is no ongoing GATT connection
+ * @retval Error code from BLE Stack if applicable
+ */
+rd_status_t ri_gatt_params_request (const ri_gatt_params_t params);
+
 #endif
```
Comment (ojousima) [inline src/interfaces/communication/ruuvi_interface_communication_ble_gatt.h:85]: Newline at the end of file

**Review by ojousima — CHANGES_REQUESTED** [review]
Please check the values against Apple guidelines https://developer.apple.com/library/archive/qa/qa1931/_index.html

**Comment by DG12** [comment]
@ojousima Why didn't Sonar cloud complain about "magic numbers" for x0006 and x0C80 ?

**Comment by ojousima** [comment]
> @ojousima Why didn't Sonar cloud complain about "magic numbers" for x0006 and x0C80 ?

SonarScanner cannot* compile code which references nRF5 SDK, so everything under nrf5_sdk15_platform is ignored by sonar scanner.

*It probably could, but after a day or two of trying to setup the nRF macro chains I gave up on Sonar Scan here

File: src/nrf5_sdk15_platform/communication/ruuvi_nrf5_sdk15_communication_ble_gatt.c (hunk 3 above)
Comment (niksiboxi) [reply src/nrf5_sdk15_platform/communication/ruuvi_nrf5_sdk15_communication_ble_gatt.c:324]: Done.

File: src/nrf5_sdk15_platform/communication/ruuvi_nrf5_sdk15_communication_ble_gatt.c (hunk 1 above)
Comment (niksiboxi) [reply src/nrf5_sdk15_platform/communication/ruuvi_nrf5_sdk15_communication_ble_gatt.c:876]: Applied a calculator created by Silicon Labs: https://docs.silabs.com/resources/bluetooth/miscellaneous/mobile/selecting-suitable-connection-parameters-for-apple-devices/source/AppleConnectionParams_R11.xlsx

File: src/nrf5_sdk15_platform/communication/ruuvi_nrf5_sdk15_communication_ble_gatt.c [hunk 6]
```diff
@@ -853,4 +859,43 @@ rd_status_t ri_gatt_dis_init (const ri_comm_dis_init_t * const p_dis)
...
+    ret_code_t err_code = NRF_SUCCESS;
+    ble_gap_conn_params_t gap_conn_params;
+    memset (&gap_conn_params, 0, sizeof (gap_conn_params));
+    gap_conn_params.slave_latency = SLAVE_LATENCY;
+    gap_conn_params.conn_sup_timeout = CONN_SUP_TIMEOUT;
+
+    switch (params)
+    {
+        case RI_GATT_TURBO:
+            gap_conn_params.min_conn_interval = 0x0006;
```
[the remark this answers is not in the document]
Comment (niksiboxi) [reply src/nrf5_sdk15_platform/communication/ruuvi_nrf5_sdk15_communication_ble_gatt.c:882]: Redefined based on Apple Documentation.

File: src/nrf5_sdk15_platform/communication/ruuvi_nrf5_sdk15_communication_ble_gatt.c (hunk 2 above)
Comment (niksiboxi) [reply src/nrf5_sdk15_platform/communication/ruuvi_nrf5_sdk15_communication_ble_gatt.c:892]: Redefined based on Apple Documentation

File: src/nrf5_sdk15_platform/communication/ruuvi_nrf5_sdk15_communication_ble_gatt.c (hunk 4 above)
Comment (niksiboxi) [reply src/nrf5_sdk15_platform/communication/ruuvi_nrf5_sdk15_communication_ble_gatt.c:100]: Restored

**Pull request merged**